### ISSUE TITLE
Image Resize Rounding Bug

### ISSUE THREAD

**Issue by mariannk** [nopSolutions/nopCommerce#1031]
Contribution: http://www.nopcommerce.com/boards/t/40616/image-resizing-bug.aspx

An issue exists in the PictureService which creates a white single pixel strip either vertical or horizontal on resized images such as thumbnails. The white strip appears on the right or bottom of the image. 

The issue appears when the original image size does not divide evenly into the destination size and the remainder is > 0.5. For example, when a final floating point width of 47.8 is determined, the remainder is dropped (not rounded) and the image is resized to only 47 pixels instead of 48 leaving a white border on the right side of the image. 

The issue is discussed here: 
http://www.nopcommerce.com/boards/t/30972/getting-one-white-pixel-along-the-height-on-picture-when-upgrading-to-33.aspx 

In my testing, the solution is to simply add Math.Round() to the results prior to casting to int. The code fixing the issue in PictureService.cs is below:
```
protected virtual Size CalculateDimensions(Size originalSize, int targetSize, 
            ResizeType resizeType = ResizeType.LongestSide, bool ensureSizePositive = true)
        {
            var newSize = new Size();
            switch (resizeType)
            {
                case ResizeType.LongestSide:
                    if (originalSize.Height > originalSize.Width)
                    {
                        // portrait 
                        newSize.Width = (int)Math.Round(originalSize.Width * (float)(targetSize / (float)originalSize.Height));
                        newSize.Height = targetSize;
                    }
                    else 
                    {
                        // landscape or square
                        newSize.Height = (int)Math.Round(originalSize.Height * (float)(targetSize / (float)originalSize.Width));
                        newSize.Width = targetSize;
                    }
                    break;
                case ResizeType.Width:
                    newSize.Height = (int)Math.Round(originalSize.Height * (float)(targetSize / (float)originalSize.Width));
                    newSize.Width = targetSize;
                    break;
                case ResizeType.Height:
                    newSize.Width = (int)Math.Round(originalSize.Width * (float)(targetSize / (float)originalSize.Height));
                    newSize.Height = targetSize;
                    break;
                default:
                    throw new Exception("Not supported ResizeType");
            }

            if (ensureSizePositive)
            {
                if (newSize.Width < 1)
                    newSize.Width = 1;
                if (newSize.Height < 1)
                    newSize.Height = 1;
            }

            return newSize;
        }
```

**Comment by AndreiMaz** [comment]
Right now just investigate it. Let's discuss before starting implementation

**Comment by skoshelev** [comment]
Example
![untitled](https://cloud.githubusercontent.com/assets/16875599/15708169/a70c5f28-2805-11e6-86b3-28402d9bbacc.png)

**Comment by skoshelev** [comment]
Closed #1031
